lib/mayaUsd/ufe/UsdTransform3dMatrixOp.cpp

```diff
@@ -0,0 +1,375 @@
+{
+    auto local = computeLocalInclusiveTransform(prim(), _op, getTime(path()));
+    GfMatrix4d unusedR, unusedP, unusedU;
+    GfVec3d    s, unusedT;
+    if (!local.Factor(&unusedR, &s, &unusedU, &unusedT, &unusedP)) {
+            TF_WARN("Cannot decompose local transform for %s", pathCStr());
+            return Ufe::Vector3d(1, 1, 1);
+    }
+    
+    return toUfe(s);
+}
+
+Ufe::TranslateUndoableCommand::Ptr UsdTransform3dMatrixOp::translateCmd(double x, double y, double z)
+{
+    return std::make_shared<UsdTranslateUndoableCmd>(
+        usdSceneItem(), _op, UsdTimeCode::Default());
+}
+
+Ufe::RotateUndoableCommand::Ptr UsdTransform3dMatrixOp::rotateCmd(
+    double x, double y, double z
+)
+{
+    return std::make_shared<UsdRotateUndoableCmd>(
+        usdSceneItem(), _op, UsdTimeCode::Default());
+}
+
+Ufe::ScaleUndoableCommand::Ptr UsdTransform3dMatrixOp::scaleCmd(
+    double x, double y, double z
+)
+{
+    return std::make_shared<UsdScaleUndoableCmd>(
+        usdSceneItem(), _op, UsdTimeCode::Default());
+}
+
+Ufe::Matrix4d UsdTransform3dMatrixOp::segmentInclusiveMatrix() const
+{
+    // Get the parent transform plus all ops including the requested one.
+    auto time = getTime(path());
+    UsdGeomXformCache xformCache(time);
+    auto parent = xformCache.GetParentToWorldTransform(prim());
+    auto local = computeLocalInclusiveTransform(prim(), _op, time);
+    return toUfe(local * parent);
+}
+ 
+Ufe::Matrix4d UsdTransform3dMatrixOp::segmentExclusiveMatrix() const
+{
+    // Get the parent transform plus all ops excluding the requested one.
+    auto time = getTime(path());
+    UsdGeomXformCache xformCache(time);
+    auto parent = xformCache.GetParentToWorldTransform(prim());
+    auto local = computeLocalExclusiveTransform(prim(), _op, time);
+    return toUfe(local * parent);
+}
+
+//------------------------------------------------------------------------------
+// UsdTransform3dMatrixOpHandler
+//------------------------------------------------------------------------------
+
+UsdTransform3dMatrixOpHandler::UsdTransform3dMatrixOpHandler(
+    const Ufe::Transform3dHandler::Ptr& nextHandler
+) : UsdTransform3dBaseHandler(), _nextHandler(nextHandler)
+{}
+
+/*static*/
+UsdTransform3dMatrixOpHandler::Ptr UsdTransform3dMatrixOpHandler::create(
+    const Ufe::Transform3dHandler::Ptr& nextHandler
+)
+{
+    return std::make_shared<UsdTransform3dMatrixOpHandler>(nextHandler);
+}
+
+Ufe::Transform3d::Ptr UsdTransform3dMatrixOpHandler::transform3d(
+    const Ufe::SceneItem::Ptr& item
+) const
+{
+    // This method can be used to edit the 3D transform of the argument, but at
+    // time of writing this is not implemented in UsdTransform3dMatrixOp, and
+    // our UsdTransform3dBaseHandler base class does not know how to edit the
+    // argument either.  Simply delegate to the next handler in the list.
+    return _nextHandler->transform3d(item);
+}
+
+Ufe::Transform3d::Ptr UsdTransform3dMatrixOpHandler::editTransform3d(
```

Comment:
This is the new handler that determines whether it can create a UsdTransform3dMatrixOp interface.  If it cannot, it delegates to the next handler.